### PR TITLE
Refactor install operator methods

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/e2e-harness/issues/25

Added generic methods for installing operators, both stable and from branch versions.

### CODE REVIEW

**Review by pipo02mix — APPROVED** [review]
lgmt

File: pkg/framework/host.go [hunk 1]
```diff
@@ -144,46 +144,51 @@ func (h *Host) DeleteGuestCluster() error {
...
 		return microerror.Mask(err)
 	}
 	defer os.Remove(tmpfile.Name())
 
-	if _, err := tmpfile.Write([]byte(awsOperatorChartValuesEnv)); err != nil {
+	if _, err := tmpfile.Write([]byte(chartValuesEnv)); err != nil {
 		return microerror.Mask(err)
 	}
 
+	cmd := fmt.Sprintf("registry install quay.io/giantswarm/%[1]s-chart%[2]s -- -n %[1]s --values %[3]s", name, version, tmpfile.Name())
```
Comment (xh3b4sd) [inline pkg/framework/host.go]: How does the formatting with the square brackets work? Never saw this before.

Comment (kopiczko) [reply pkg/framework/host.go]: I also had to educate myself: https://golang.org/pkg/fmt/ -> search for `Explicit argument indexes`.

**Review by indyfree — APPROVED** [review]
Nice

File: pkg/framework/host.go (hunk 1 above)
Comment (fgimenez) [reply pkg/framework/host.go]: Exactly, here, given that the name is used twice in the format string, the goal is not having to repeat values, see https://play.golang.org/p/IfqFTpiV5eh for example in contrast to https://play.golang.org/p/gnzXPMw2J5F

**Comment by fgimenez** [comment]
I removed the now unused cert-operator constants too. Going to merge now, there is still some aws specific code related to the CR chart and patching mainly, will work on that next as part of https://github.com/giantswarm/e2e-harness/issues/25. We can iterate on the operator code changes in this PR if needed.

**Pull request merged**